### PR TITLE
Update 'css FAQ' and fix prettier

### LINKED ISSUE #14512: Page 'CSS FAQ' was not fully prettified
### MDN URL

https://developer.mozilla.org/zh-CN/docs/Learn/CSS/Howto/CSS_FAQ

### 与此议题相关的小节或标题？

_No response_

### 哪些信息是错误、无用或不完善的？

We are running prettier on our document, and for this document, prettier seems to break the link. Also, the content are also outdated, an update is needed for the document.
ref:
https://github.com/mdn/translated-content/pull/14492/files#r1271250855

### 你期望看到什么？

Update the document and remove this file out of prettierignore.

### 你有什么支持的链接、参考或引用？

_No response_

### 你还想补充什么内容？

I would like to do the update work 🤔 

### MDN metadata

<details>
<summary>Page report details</summary>

* Folder: `zh-cn/learn/css/howto/css_faq`
* MDN URL: https://developer.mozilla.org/zh-CN/docs/Learn/CSS/Howto/CSS_FAQ
* GitHub URL: https://github.com/mdn/translated-content/blob/main/files/zh-cn/learn/css/howto/css_faq/index.md
* Last commit: https://github.com/mdn/translated-content/commit/e253de6751aaa63fc7c8163e86a1a60bcb17e73f
* Document last modified: 2022-09-27T08:07:17.000Z

</details>

### PR DESCRIPTION
**Pull request by jasonren0403**
### Description

Full update 'css faq' and remove this out of prettierignore.

### Related issues and pull requests

closes #14512

### CODE REVIEW

File: files/zh-cn/learn/css/howto/css_faq/index.md [hunk 1]
```diff
@@ -1,27 +1,31 @@
...
 
 现代浏览器主要有两种渲染模式：
 
-- _怪异模式_：又称向后兼容模式，允许将传统网页渲染为作者意图。旧浏览器使用的非标准渲染规则。不完整的、不正确的、缺少 DOCTYPE 声明或已知的 DOCTYPE 声明中普遍使用 2001 年以前的文件将在怪异模式中呈现。
-- _标准模式_：浏览器试图严格遵守 W3C 标准。新 HTML 网页有望被设计为符合标准的浏览器，这样做的结果就是，用现代 DOCTYPE 声明的页面将被用标准模式呈现。
+- _怪异模式_：又称向后兼容模式，允许按照旧版浏览器使用的非标准呈现规则，按作者的意图呈现旧版网页。如果文档中存在不完整、不正确或缺失的 `DOCTYPE` 声明，或已知的 2001 年以前常用的 `DOCTYPE` 声明，则会以怪异模式呈现。
+- _标准模式_：浏览器试图严格遵守 W3C 标准。新的 HTML 页面预计是为符合标准的浏览器设计的，因此，带有现代 `DOCTYPE` 声明的页面将使用标准模式呈现。
 
-基于 Gecko 的浏览器，有三分之一 _[Almost Standards Mode](/zh-CN/docs/Gecko's_%22Almost_Standards%22_Mode)_，只有一些小怪癖。
+有三分之一的基于 Gecko 的浏览器处于*[准标准模式](/zh-CN/docs/Mozilla/Gecko_Almost_Standards_Mode)*，只有一些小怪癖。
```
Comment (jasonren0403) [inline files/zh-cn/learn/css/howto/css_faq/index.md:19]: should we just remove the link 🤔 seems that it is not going to be documented

Comment (jasonren0403) [reply files/zh-cn/learn/css/howto/css_faq/index.md:19]:
```suggestion
有三分之一的基于 Gecko 的浏览器处于[*准标准模式*](/zh-CN/docs/Mozilla/Gecko_Almost_Standards_Mode)，只有一些小怪癖。
```

Comment (yin1999) [reply files/zh-cn/learn/css/howto/css_faq/index.md:19]: > has only a few minor quirks.

只是有一些怪异行为。

https://zh.wikipedia.org/zh-hans/%E6%80%AA%E5%BC%82%E6%A8%A1%E5%BC%8F

Comment (yin1999) [reply files/zh-cn/learn/css/howto/css_faq/index.md:19]: > should we just remove the link 🤔 seems that it is not going to be documented

+1 for removing this 👍

Comment (jasonren0403) [reply files/zh-cn/learn/css/howto/css_faq/index.md:19]: PR proposed https://github.com/mdn/content/pull/28163

File: files/zh-cn/learn/css/howto/css_faq/index.md [hunk 2]
```diff
@@ -1,27 +1,31 @@
...
 
-浏览器使用 DOCTYPE 声明来选择是否使用更符合 Web 标准或兼容旧浏览器的 bug 的模式。在你的 HTML 的开始使用一个正确的和现代的 DOCTYPE 声明将改善浏览器标准执行。
+浏览器使用 `DOCTYPE` 声明来选择是否使用更符合 Web 标准或兼容旧浏览器缺陷的模式。在你的 HTML 的开始使用一个正确的和现代的 `DOCTYPE` 声明将改善浏览器标准执行。
 
 现代浏览器主要有两种渲染模式：
 
-- _怪异模式_：又称向后兼容模式，允许将传统网页渲染为作者意图。旧浏览器使用的非标准渲染规则。不完整的、不正确的、缺少 DOCTYPE 声明或已知的 DOCTYPE 声明中普遍使用 2001 年以前的文件将在怪异模式中呈现。
-- _标准模式_：浏览器试图严格遵守 W3C 标准。新 HTML 网页有望被设计为符合标准的浏览器，这样做的结果就是，用现代 DOCTYPE 声明的页面将被用标准模式呈现。
+- _怪异模式_：又称向后兼容模式，允许按照旧版浏览器使用的非标准呈现规则，按作者的意图呈现旧版网页。如果文档中存在不完整、不正确或缺失的 `DOCTYPE` 声明，或已知的 2001 年以前常用的 `DOCTYPE` 声明，则会以怪异模式呈现。
+- _标准模式_：浏览器试图严格遵守 W3C 标准。新的 HTML 页面预计是为符合标准的浏览器设计的，因此，带有现代 `DOCTYPE` 声明的页面将使用标准模式呈现。
```
Comment (yin1999) [inline files/zh-cn/learn/css/howto/css_faq/index.md:17]:
```suggestion
- _怪异模式_：又称向后兼容模式，允许按照旧版浏览器使用的非标准渲染规则，按作者的意图呈现旧版网页。如果文档中存在不完整、不正确或缺失的 `DOCTYPE` 声明，或已知的 2001 年以前常用的 `DOCTYPE` 声明，则会以怪异模式渲染。
- _标准模式_：浏览器试图严格遵守 W3C 标准。新的 HTML 页面预计是为符合标准的浏览器设计的，因此，带有现代 `DOCTYPE` 声明的页面将使用标准模式渲染。
```

File: files/zh-cn/learn/css/howto/css_faq/index.md [hunk 3]
```diff
@@ -245,35 +243,34 @@ div {
...
-transform: rotate(90deg);
+-webkit-text-stroke: 4px navy;
+text-stroke: 4px navy;
 ```
 
-> **备注：** 为更深层次了解处理前缀属性，请参阅 [跨浏览器测试](/zh-CN/docs/Learn/Tools_and_testing/Cross_browser_testing) 模块的 [处理常见的 HTML 和 CSS 问题](/zh-CN/docs/Learn/Tools_and_testing/Cross_browser_testing/HTML_and_CSS)。
+> **备注：** 为更深层次了解处理前缀属性，请参阅[跨浏览器测试](/zh-CN/docs/Learn/Tools_and_testing/Cross_browser_testing)模块的[处理常见的 HTML 和 CSS 问题——处理 CSS 前缀](/zh-CN/docs/Learn/Tools_and_testing/Cross_browser_testing/HTML_and_CSS#处理_css_前缀)。
 
-> **备注：** 为了解 Mozilla 前缀属性 CSS 的更多信息，请参阅 [Mozilla CSS 扩展](/zh-CN/docs/Web/CSS/Mozilla_Extensions)。
+> **备注：** 请参阅 [Mozilla CSS 扩展](/zh-CN/docs/Web/CSS/Mozilla_Extensions)、[Microsoft CSS 扩展](/zh-CN/docs/Web/CSS/Microsoft_Extensions)和 [WebKit CSS 扩展](/zh-CN/docs/Web/CSS/WebKit_Extensions) 以了解浏览器前缀 CSS 属性的列表。
```
[the remark this answers is not in the document]
Comment (jasonren0403) [reply files/zh-cn/learn/css/howto/css_faq/index.md:268]: +1 for removing it 👍 Could you help me to propose a PR?

[the remark this answers is not in the document]
Comment (yin1999) [reply files/zh-cn/learn/css/howto/css_faq/index.md:268]: filled: mdn/content#2821

File: files/zh-cn/learn/css/howto/css_faq/index.md [hunk 4]
```diff
@@ -1,27 +1,31 @@
...
-这是最常用的触发标准模式或准标准模式的 DOCTYPE 声明列表：
+这是最常用的触发标准模式的 `DOCTYPE` 声明：
 
 ```html
-<!doctype html> /* 这一行是 HTML5 的 doctype 声明。使用该声明会使现代浏览器使用
-HTML5 解析器处理页面，这是推荐的 doctype 声明。*/
+<!doctype html>
 ```
 
+你应该尽可能使用上述 doctype 声明。还有其他有效的传统 doctype 说明，它们会触发标准模式或准标准模式：
```
Comment (yin1999) [inline files/zh-cn/learn/css/howto/css_faq/index.md:27]:
```suggestion
你应该尽可能使用上述 doctype 声明。还有其他有效的传统 doctype 说明，它们会触发标准模式或接近标准模式：
```

File: files/zh-cn/learn/css/howto/css_faq/index.md [hunk 5]
```diff
@@ -38,30 +42,24 @@ HTML5 解析器处理页面，这是推荐的 doctype 声明。*/
...
-较少样式的样式表通常性能更高。因此建议尽可能多地使用类，保留 id 作为特定用途（比如链接 label 标签和 form 元素或者为语义上唯一的元素应用样式）：
+一般建议尽量使用类，只有在特定用途（如连接标签和表单元素或用于必须在语义上唯一的样式元素）绝对必要时才使用 id：
 
 - 使用类名可以让样式具有可扩展性——即使目前只有一个元素使用这个规则集来定义样式，未来可能会添加更多。
-
 - 类名可以让你同时为多个元素赋予样式，减少样式表的大小，避免了为每一个 id 选择器撰写同样的样式信息。越小的样式表带来的性能体验也就越好。
-
 - 类名选择器比 id 选择器的[优先级](/zh-CN/docs/Learn/CSS/Building_blocks/Cascade_and_inheritance#优先级)更低，所以可以很方便地覆盖它们。
 
 > **备注：** 在这篇文档中更深入层次的了解 [CSS 选择器](/zh-CN/docs/Learn/CSS/Building_blocks/Selectors)。
```
Comment (yin1999) [inline files/zh-cn/learn/css/howto/css_faq/index.md:65]:
```suggestion
> **备注：** 参见 [CSS 选择器](/zh-CN/docs/Learn/CSS/Building_blocks/Selectors)，以了解详情。
```

File: files/zh-cn/learn/css/howto/css_faq/index.md [hunk 6]
```diff
@@ -245,35 +243,34 @@ div {
...
-transform: rotate(90deg);
+-webkit-text-stroke: 4px navy;
+text-stroke: 4px navy;
 ```
 
-> **备注：** 为更深层次了解处理前缀属性，请参阅 [跨浏览器测试](/zh-CN/docs/Learn/Tools_and_testing/Cross_browser_testing) 模块的 [处理常见的 HTML 和 CSS 问题](/zh-CN/docs/Learn/Tools_and_testing/Cross_browser_testing/HTML_and_CSS)。
+> **备注：** 为更深层次了解处理前缀属性，请参阅[跨浏览器测试](/zh-CN/docs/Learn/Tools_and_testing/Cross_browser_testing)模块的[处理常见的 HTML 和 CSS 问题——处理 CSS 前缀](/zh-CN/docs/Learn/Tools_and_testing/Cross_browser_testing/HTML_and_CSS#处理_css_前缀)。
 
-> **备注：** 为了解 Mozilla 前缀属性 CSS 的更多信息，请参阅 [Mozilla CSS 扩展](/zh-CN/docs/Web/CSS/Mozilla_Extensions)。
+> **备注：** 请参阅 [Mozilla CSS 扩展](/zh-CN/docs/Web/CSS/Mozilla_Extensions) 和 [WebKit CSS 扩展](/zh-CN/docs/Web/CSS/WebKit_Extensions)，以了解浏览器前缀 CSS 属性的列表。
```
Comment (yin1999) [inline files/zh-cn/learn/css/howto/css_faq/index.md:268]:
```suggestion
> **备注：** 请参阅 [Mozilla CSS 扩展](/zh-CN/docs/Web/CSS/Mozilla_Extensions)和 [WebKit CSS 扩展](/zh-CN/docs/Web/CSS/WebKit_Extensions)，以了解浏览器前缀 CSS 属性的列表。
```

**Review by yin1999 — APPROVED** [review]
Thanks. Just some suggestions. Then feel free to merge this :)

**Pull request merged**